project_generator/project.py

```diff
@@ -427,71 +399,41 @@ def generate_dict_for_tool(self, tool):
             if self.tools.get_value(tool, 'toolchain') != tool_spec:
                 tool_specific_settings.append(self.tool_specific[tool_spec])
 
-        d = {
-            'name': self.name,
-            'mcu': self.mcu,
-            'core': self.core,
-            'target': self.target,
-            'output_type': self.output_type,
-            'build_dir': self.build_dir,
-            'tools_supported': self.tools_supported,
-            'debugger': self.debugger,
-            'output_dir': self.output_dir,
-            'includes':
-                self.include_paths + list(flatten([settings.include_paths for settings in tool_specific_settings])),
-            'source_paths':
-                self.source_paths + list(flatten([settings.source_paths for settings in tool_specific_settings])),
-
-            'source_files':
-                merge_recursive(self.source_groups,
+        self.project['includes'] =  self.project['includes'] + list(flatten([settings.include_paths for settings in tool_specific_settings]))
+        self.project['source_paths'] =  self.project['source_paths'] + list(flatten([settings.include_paths for settings in tool_specific_settings]))
+        self.project['source_files'] = merge_recursive(self.source_groups,
                                 {k: v for settings in tool_specific_settings for k, v in settings.source_groups.items()},
-                                toolchain_specific_settings.source_groups),
-            # for backwards compatibility
-            'source_files_c': self.format_source_files('c',tool_specific_settings, toolchain_specific_settings),
-
-            'source_files_cpp': self.format_source_files('cpp',tool_specific_settings, toolchain_specific_settings),
-
-            'source_files_s':  self.format_source_files('s',tool_specific_settings, toolchain_specific_settings),
+                                toolchain_specific_settings.source_groups)
+        for ext in ["c","cpp","s","lib"]:
+           key = "source_files_"+ext
+           self.project[key] = self.format_source_files(ext, tool_specific_settings, toolchain_specific_settings)
 
-            'source_files_obj': merge_recursive(self.format_source_files('obj',tool_specific_settings, toolchain_specific_settings),
-                                                self.format_source_files('o',tool_specific_settings, toolchain_specific_settings)),
+        self.project['source_files_obj']= merge_recursive(self.format_source_files('obj',tool_specific_settings, toolchain_specific_settings),
```

Comment:
I want to do something like this for each extension, iterate over the extensions mapped in FILE_EXTENSIONS and merge them all, but having trouble with syntax